src/beacon-api/app/services/grid_sync_service.py

```
@@ -0,0 +1,571 @@
+
+    all_grids: List[Dict[str, Any]] = []
+    page = 1
+    actual_limit = limit
+
+    async with httpx.AsyncClient(timeout=60.0, follow_redirects=True) as client:
+        while True:
+            try:
+                data = await _fetch_grid_page(
+                    client, url, headers, page, (page - 1) * actual_limit, limit
+                )
+            except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)